server/src/main/java/org/elasticsearch/cluster/node/DiscoveryNode.java

```diff
@@ -488,6 +488,16 @@ public Version getVersion() {
         return this.versionInfo.nodeVersion();
     }
 
+    public int getLegacyVersionIdOrThrow() {
```

Comment:
Could return `OptionalInt` here, leaving it up to the caller to determine what should be done if its out of bounds